csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -64,7 +64,6 @@ internal IMessage CreateTemplate()
         /// <returns>The newly parsed message.</returns>
         public IMessage ParseFrom(byte[] data)
         {
-            ProtoPreconditions.CheckNotNull(data, "data");
```

Comment:
The downside of removing this is that the parameter name will be wrong in the exception (buffer rather than data). But I'm not massively worried.